src/frontend/sharedComponents/ActionRow.tsx

```diff
@@ -51,6 +51,10 @@ export const ActionsRow = ({fieldRefs}: ActionButtonsProps) => {
     openOnMount: false,
   });
 
+  const [pendingAction, setPendingAction] = useState<'navigateToAudio' | null>(
+    null,
+  );
```

Comment:
I think it is clearer to use a boolean instead of creating 2 opposing states. Eg `const [shouldNavigateToAudio, setShouldNavigateToAudio] = useState(false)`